### ISSUE TITLE
If you want to delete your profile you should have to type in your current password

### ISSUE THREAD

**Issue by ovalseven8** [qTox/qTox#2338]
For more security

**Comment by aaannndddyyy** [comment]
Sounds reasonable at first.
But what should it protect against?

If you want to protect the user from accidentally deleting his profile, a simple "Are you sure?" Pop up dialogue is enough.

What happens, if you forget your password? Then the profile is useless.
You can either keeping it, hoping you will remember the password at a later time, or you may want to get rid of the useless profile - without having to know its forgotten password.

The only thing it would protect against  is when other persons get access to your device. In that case they could also format your hard disk, so this protection is no protection at all.
The password is there to keep the chat contents, the tox identity and contact list confidential but not to protect against deletion of something.

**Comment by TheNain38** [comment]
@ovalseven8 Yes, you can't prevent someone from deleting files and, so tox profiles too

**Comment by ProMcTagonist** [comment]
> If you want to protect the user from accidentally deleting his profile, a simple "Are you sure?" Pop up dialogue is enough.

No. 

https://elementary.io/docs/human-interface-guidelines#always-provide-an-undo

**Comment by aaannndddyyy** [comment]
Then provide an undo button. But typing the password does not accomplish what you want. You would need to move 'deleted' profiles to an internal trash bin . . . 
Needlessly typing a password is a nuissance, and users want to easily get rid of old unused profiles.

**Comment by An-dy** [comment]
If you want get sure that a profile don't get deleted, the solution is simple with Linux: (ext2-4 filesystem)
chattr +i /home/user/.config/tox/ProfileName.tox    (un-immutable with chattr -i )
And the profile is immutable from delete.

P.S. Btw. interesting reaction from qtox:
src/persistence/profile.cpp:299 : Debug: Saving tox save to  "/home/user/.config/tox/ProfileName.tox"
src/persistence/profile.cpp:303 : Critical: Tox save file  "/home/user/.config/tox/ProfileName.tox"  couldn't be opened

**Comment by aaannndddyyy** [comment]
the file needs to be writeable

**Comment by An-dy** [comment]
You are right. I mixed it up with the +a attribute.
"The “a” stands for “append” and this flag indicates that a file can only be opened for append operations, i.e. additional content can only be added to the file but none of the existing content can be modified." 
It works. You can change things and they get saved but you can't delete it. I checked it.
